controllers/job_controller_test.go

```diff
@@ -0,0 +1,158 @@
+package controllers_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
```

Comment:
I'm concerned with a couple things here.

- The dot import. _"Don't use the import . notation, which can simplify tests that must run outside the package they are testing, but should otherwise be avoided."_ --[ref](https://go.dev/doc/effective_go#:~:text=Don%27t%20use%20the%20import%20.%20notation%2C%20which%20can%20simplify%20tests%20that%20must%20run%20outside%20the%20package%20they%20are%20testing%2C%20but%20should%20otherwise%20be%20avoided.)
- The use of this assert library. _"Avoid the use of 'assert' libraries to help your tests"_ --[ref](https://github.com/golang/go/wiki/TestComments#assert-libraries) With this one, i see that `ginko` and `gomega` are already used a bunch through out the repo. 

I won't block on using the assert libraries but the dot import feels like it should be avoided.

Reply:
> The use of this assert library

I really think that code review point is clutching at straws and would strongly encourage ignoring it. The assumptions it makes for its rationale are inaccurate and its alternative is hard to justify as 'better' given in practice tests aren't as clear cut and lots of branch/predicate logic can be hard to parse.

Reply:
> The dot import

I took another look at this as, usually, I'd agree. Similar to the above though, if applied broadly without consideration of context you end up being uncomplimentary to the style. Specifically, Ginkgo and Gomega are tools for BDD. BDD attempts to use natural language to model expectations. Applying the dot import guideline results in breaks when constructing assertions which defeats the point. 

I'm not fond of BDD and would prefer we remove this entirely - I might do just that on a separate PR. I'll update this given I intend on removing it anyway but I wanted to call it out nonetheless.